src/app/file/svg_format.cpp

```diff
@@ -0,0 +1,149 @@
+// Aseprite
+// Copyright (c) 2018  Igara Studio S.A.
+//
+// This program is distributed under the terms of
+// the End-User License Agreement for Aseprite.
+//
+
+#ifdef HAVE_CONFIG_H
+#include "config.h"
+#endif
+
+#include "app/file/file.h"
+#include "app/file/file_format.h"
+#include "app/file/format_options.h"
+#include "base/cfile.h"
+#include "base/file_handle.h"
+#include "doc/doc.h"
+
+namespace app {
+
+using namespace base;
+
+class SvgFormat : public FileFormat {
+  const char* onGetName() const override {
+    return "svg";
+  }
+
+  void onGetExtensions(base::paths& exts) const override {
+    exts.push_back("svg");
+  }
+
+  dio::FileFormat onGetDioFormat() const override {
+    return dio::FileFormat::SVG_IMAGE;
+  }
+
+  int onGetFlags() const override {
+    return
+      FILE_SUPPORT_SAVE |
+      FILE_SUPPORT_RGB |
+      FILE_SUPPORT_RGBA |
+      FILE_SUPPORT_GRAY |
+      FILE_SUPPORT_INDEXED |
+      FILE_SUPPORT_SEQUENCES;
+  }
+
+  bool onLoad(FileOp* fop) override;
+#ifdef ENABLE_SAVE
+  bool onSave(FileOp* fop) override;
+#endif
+};
+
+FileFormat* CreateSvgFormat()
+{
+  return new SvgFormat;
+}
+
+  bool SvgFormat::onLoad(FileOp* fop) { return false;}
+
+#ifdef ENABLE_SAVE
+
+bool SvgFormat::onSave(FileOp* fop)
+{
+  const Image* image = fop->sequenceImage();
+  unsigned char image_palette[256][4];
+  int x, y, c, r, g, b, a, alpha;
+  float opacity;
+  bool need_pal = (image->pixelFormat() == IMAGE_INDEXED)? true: false;
+  FileHandle handle(open_file_with_exception_sync_on_close(fop->filename(), "wb"));
+  FILE* f = handle.get();
+
+  if (need_pal) {
+    for (y=0; y<256; y++) {
+      fop->sequenceGetColor(y, &r, &g, &b);
+      image_palette[y][0] = r;
+      image_palette[y][1] = g;
+      image_palette[y][2] = b;
+      fop->sequenceGetAlpha(y, &a);
+      image_palette[y][3] = a;
+    }
+  }
+  fprintf(f, "<?xml version=\"1.0\" encoding=\"UTF-8\" ?>\n");
+  fprintf(f, "<svg version=\"1.1\" width=\"%d\" height=\"%d\" xmlns=\"http://www.w3.org/2000/svg\">\n",
+          image->width(), image->height());
+  switch (image->pixelFormat()) {
+
+    case IMAGE_RGB:
+      for (y=0; y<image->height(); y++) {
+        for (x=0; x<image->width(); x++) {
+          c = get_pixel(image, x, y);
```

Comment:
We can use the fast versions of get_pixel() on each case (something the tga_format.cpp is not using 😢)
```suggestion
          c = get_pixel_fast<RgbTraits>(image, x, y);
```